### PR TITLE
Delete RooCintUtils.h and RooCintUtils.cxx

### PR DESCRIPTION
**Pull request by yamaguchi1024**
I know this is not a small change, so any comments are welcome!

Background:
RooCintUtils was providing a wrapper between Cint and roofit. This implementation is very old, and is
using Cint iteration interface a lot. Recently we were trying to kill
RooCintUtils code where its using iteration and to change them to new
Cling interface in order to improve the performance.

RooCintUtils is an utility class, so this is internal namespace which is only used inside roofit.
Since there are just 3 methods left, this RooCintUtils is no longer
worth spending a file. Let's delete this and migrate methods to
RooFactoryWSTool.

### CODE REVIEW

**Comment by dpiparo** [comment]
@yamaguchi1024 : nice. I let the reviewers comment but I support this change.

**Comment by Axel-Naumann** [comment]
Great work, thanks, @yamaguchi1024 !

This seems to be ok (but needs to be reviewed by @lmoneta ).

For future PRs I'd appreciate keeping functional changes (enum iteration) and non-functional changes (moving functions into a different source file) in separate commits. As is I have a hard time seeing where the functional changes are taking place. So much so that I trust Vassil to have reviewed the functional part, such that I don't need to sign off on it.

Thanks!

**Comment by yamaguchi1024** [comment]
Thanks for comments!
@phsft-bot This is actually depending on #1722, I have to fix this first

File: roofit/roofitcore/src/RooFactoryWSTool.cxx [hunk 1]
```diff
@@ -228,7 +228,84 @@ RooCategory* RooFactoryWSTool::createCategory(const char* name, const char* stat
   return _ws->cat(name) ;
 }
 
+bool isEnum(const char* classname) {
```
Comment (pcanal) [inline roofit/roofitcore/src/RooFactoryWSTool.cxx]: If those routines are only for use inside this file they should be either prefix with R__ or placed in an anonymous namespace.  Either way, they also need to be declared static.  thanks.

**Pull request merged**